minimailgun/api.py

```diff
@@ -0,0 +1,96 @@
+
+import os
+import itertools
+import traceback
+from flask import Flask, Blueprint, jsonify, request, abort, current_app
+from werkzeug.exceptions import default_exceptions, HTTPException
+
+from minimailgun.store import store, MailLookupError, UnablToAddMessageError
+from minimailgun.tasks import handle_new_message
+
+
+mailgun_api = Blueprint('mini-mailgun', __name__)
+
+
+@mailgun_api.route('/mail', methods=['POST'])
+def create_mail():
+    if request.content_type != 'application/json':
```

Comment:
These two checks look generic enough so they may be useful across all handlers (routes) of the application, if we want to make the API consistent (which we should).

How about we create our own decorator that wraps `mailgun_api.route` and adds generic things that apply to all API methods (like these checks)? This will also free handlers from the noise, make them easier to read/understand.